icu4c/source/i18n/number_rounding.cpp

```diff
@@ -278,23 +278,23 @@ Precision IncrementPrecision::withMinFraction(int32_t minFrac) const {
 }
 
 FractionPrecision Precision::constructFraction(int32_t minFrac, int32_t maxFrac) {
-    FractionSignificantSettings settings;
+    FractionSignificantSettings settings{};
```

Comment:
This seems fine, but why isn't it the same as the old `FractionSignificantSettings settings;`?
Either one should invoke the default constructor, right?

Reply:
> FractionSignificantSettings is a struct, not a class

... but these are the same thing, except that struct members default to public rather than private.

> https://en.cppreference.com/w/c/language/struct_initialization

That's for C, not for C++.

For C++ I found https://en.cppreference.com/w/cpp/language/aggregate_initialization
In particular the section “Implicitly initialized elements” there.

> FractionSignificantSettings settings{};
> 
> initializing the object of struct , which call the default constructor.

I was confused because there is no default constructor, and there are no [default member initializers](https://en.cppreference.com/w/cpp/language/data_members#Member_initialization).
The aggregate_initialization page says “Otherwise, if the element is not a reference, the element is copy-initialized from an empty initializer list.”

So these fields get whatever null values their types have.

I find this difference between using and not using `{}` confusing -- but it's not the first thing I find confusing about modern C++.